src/Features/Core/Portable/EditAndContinue/EditAndContinueDiagnosticUpdateSource.cs

```diff
@@ -64,56 +64,53 @@ public void ReportDiagnostics(Workspace workspace, Solution solution, ProjectId?
                 return;
             }
 
-            using var _1 = ArrayBuilder<DiagnosticData>.GetInstance(out var documentDiagnosticData);
-            using var _2 = ArrayBuilder<DiagnosticData>.GetInstance(out var nonDocumentDiagnosticData);
-            var options = solution.Options;
-            var project = (projectId != null) ? solution.GetProject(projectId) : null;
+            var documentDiagnostics = diagnostics.WhereAsArray(d => d.DocumentId != null);
+            var projectDiagnostics = diagnostics.WhereAsArray(d => d.DocumentId == null && d.ProjectId != null);
+            var solutionDiagnostics = diagnostics.WhereAsArray(d => d.DocumentId == null && d.ProjectId == null);
```

Comment:
what's an example of a solution diagnostic?

Reply:
The debugger can't do EnC because the process doesn't support it.

Reply:
It's not tied to a specific project.